### PR TITLE
Implement workspace/willRenameFiles for single-level file moves

### LINKED ISSUE #6780: LSP 3.16: Use workspace/willRenameFiles and/or worskspace/didRenameFiles to automatically rename modules
Once #6779 is in we will gain the ability to use these (and their create/delete variants) to detect file operations in supporting clients.  We should use these to do things like: client has renamed foo.rs to bar.rs so update the `foo` mod to be `bar`.

### PR DESCRIPTION
**Pull request by kjeremy**
Automatically rename modules during file rename if they're in the same directory.

Fixes #6780

### CODE REVIEW

**Comment by kjeremy** [comment]
scratch-1608680004870

File: crates/ide/src/lib.rs [hunk 1]
```diff
@@ -535,6 +535,14 @@ impl Analysis {
         self.with_db(|db| references::rename::prepare_rename(db, position))
     }
 
+    pub fn will_rename_file(
+        &self,
+        file_id: FileId,
+        new_name: &str,
```
Comment (matklad) [inline crates/ide/src/lib.rs:541]:
```suggestion
        new_name_stem: &str,
```

or some equivalent name to make it clear that `.rs` is not included.

**Review by matklad — COMMENTED** [review]
LGTM!

**Pull request merged**